faf-commons-data/src/main/java/com/faforever/commons/replay/ReplayDataParser.java

```diff
@@ -51,12 +53,12 @@ public ReplayDataParser(Path path) {
   }
 
   private String readString(LittleEndianDataInputStream dataStream) throws IOException {
-    StringBuilder stringBuilder = new StringBuilder();
-    char character;
-    while ((character = (char) dataStream.readByte()) != 0) {
-      stringBuilder.append(character);
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    byte tempByte;
+    while ((tempByte =  dataStream.readByte()) != 0) {
```

Comment:
I saw this https://github.com/google/guava/blob/master/guava/src/com/google/common/io/LittleEndianDataInputStream.java#L176

Is it possible to use in our case?

Reply:
I don't know the multi-byte utf, so i'm not sure, if we can read until value in not 0.

Reply:
Anyway, can we read use `readUTF` for our case? There should be some sort of implementation: https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/io/DataInputStream.java#L588
I think, we won't need to have loop afterwards.

Reply:
I`m tried to use this method but all I have is 

> "Caused by: java.io.UTFDataFormatException: malformed input around byte 117"

As I understand readUTF read whole stream (into one String) but in this case we need to read stream part by part (one stream is passed many times in this method and loop checking that data is still available).

Reply:
Ok, for me that one what you've done makes now sense

Reply:
Good job ;-)